provider/aws/aws_test.go

```diff
@@ -836,27 +935,33 @@ func TestAWSBatchChangeSet(t *testing.T) {
 }
 
 func TestAWSBatchChangeSetExceeding(t *testing.T) {
-	var cs []*route53.Change
+	var cs Route53Changes
 	const testCount = 50
 	const testLimit = 11
 	const expectedBatchCount = 5
 	const expectedChangesCount = 10
 
 	for i := 1; i <= testCount; i += 2 {
-		cs = append(cs, &route53.Change{
-			Action: aws.String(route53.ChangeActionCreate),
-			ResourceRecordSet: &route53.ResourceRecordSet{
-				Name: aws.String(fmt.Sprintf("host-%d", i)),
-				Type: aws.String("A"),
+		cs = append(cs,
+			&Route53Change{
```

Comment:
There's a lot of places in this PR where the only change is that a `route53.Change` is wrapped by our custom `Route53Change{Change: x}`. If we could separate this change from the actually meaningful part (handling error and retry) then the PR would get much smaller.

We could isolate this change in a single commit and the rest in another commit. Or we just create another PR where we only do the `Route53Change` wrapping (without changing any behaviour) and rebase this PR onto it.

Reply:
I agree it would be a good idea to do the replacement as @linki is suggesting.

Reply:
Done.